### PR TITLE
New argument --incident-id to "run" command

### PR DESCRIPTION
**Pull request by ennozdd**
## Related Issues
fixes:

## Description

Run command is only able to execute commands in the playground, with --incident-id argument users would be able to execute in any incident

image
image

### CODE REVIEW

**Comment by thefrieddan1** [comment]
Hi @ennozdd,

Thank you for your contribution.
I have modified a unit test to cover the addition of the new argument.
But I dont have permissions to push to this branch.
Can you please grant me permissions or if you like to add it yourself.

Thanks.

**Comment by ennozdd** [comment]
@thefrieddan1 I added you as a collaborator to my forked repository, you should be able to push to my branch

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 * Fixed the **generate-test-playbook** which failed on an unexpected keyword argument - 'console_log_threshold'.
 * Fixed an issue where **prepare-content** would not properly parse the `fromVersion` and `toVersion` attributes of XSIAM-Dashbaord and XSIAM-Report content items.
 * Fixed an issue where **validate** command did not fail on non-existent dependency ids of non-mandatory dependant content.
+* Added *--incident-id* argument to *run* command.
```
Comment (dorschw) [inline CHANGELOG.md:12]:
```suggestion
* Added the `--incident-id` argument (optional) to the **run** command.
```

File: demisto_sdk/commands/run_cmd/runner.py [hunk 2]
```diff
@@ -65,10 +67,12 @@ def __init__(
 
     def run(self):
         """Runs an integration command on Demisto and prints the result."""
-        playground_id = self._get_playground_id()
+        investigation_id = (
+            self.incident_id if self.incident_id else self._get_playground_id()
```
Comment (dorschw) [inline demisto_sdk/commands/run_cmd/runner.py:71]:
```suggestion
            self.incident_id or self._get_playground_id()
```

File: demisto_sdk/tests/integration_tests/run_integration_test.py [hunk 3]
```diff
@@ -115,28 +115,26 @@ def test_json_to_outputs_flag_fail_no_prefix(
...
     mocker.patch.object(DefaultApi, "download_file", return_value=DEBUG_FILE_PATH)
     # mock to set prefix instead of getting it from input
 
-    command = "!kl-get-records"
-    run_result = CliRunner(
-        mix_stderr=False,
-    ).invoke(main, ["run", "-q", command, "--json-to-outputs"])
-    assert 1 == run_result.exit_code
+    run_result = runner.run()
+    assert 1 == run_result
```
Comment (dorschw) [inline demisto_sdk/tests/integration_tests/run_integration_test.py:137]:
```suggestion
    assert runner.run() == 1
```

File: demisto_sdk/tests/integration_tests/run_integration_test.py [hunk 4]
```diff
@@ -115,28 +115,26 @@ def test_json_to_outputs_flag_fail_no_prefix(
...
     - Ensure the json_to_outputs command is failing due to no prefix argument provided.
     """
     logger_info = mocker.patch.object(logging.getLogger("demisto-sdk"), "info")
     monkeypatch.setenv("COLUMNS", "1000")
     # mocks to allow the command to run locally
-    mocker.patch.object(Runner, "_get_playground_id", return_value="pg_id")
-    mocker.patch.object(Runner, "_run_query", return_value=["123"])
+    runner = Runner(query="kl-get-records", json_to_outputs=True, incident_id="pg_id")
+    mocker.patch.object(runner, "_run_query", return_value=["123"])
     # mock to get test log file
```
Comment (dorschw) [inline demisto_sdk/tests/integration_tests/run_integration_test.py:132]: please add a test (using mocker) that the incident_id value is passed as expected

File: demisto_sdk/commands/run_cmd/runner.py [hunk 5]
```diff
@@ -65,10 +67,11 @@ def __init__(
 
     def run(self):
         """Runs an integration command on Demisto and prints the result."""
-        playground_id = self._get_playground_id()
+        investigation_id = self.incident_id or self._get_playground_id()
+        logger.info(f"run investigation_id is: {investigation_id}")
```
Comment (dorschw) [inline demisto_sdk/commands/run_cmd/runner.py:71]:
```suggestion
        logger.debug(f"running command in {investigation_id=}")
```

File: demisto_sdk/tests/integration_tests/run_integration_test.py [hunk 6]
```diff
@@ -141,3 +141,41 @@ def test_json_to_outputs_flag_fail_no_prefix(
...
+    run_result = CliRunner(
+        mix_stderr=False,
+    ).invoke(main, ["run", "-q", command, "--incident-id", "pg_id"])
+
+    assert run_result.exit_code == 0
+    assert not run_result.stderr
+    assert not run_result.exception
+
+    assert str_in_call_args_list(
+        logger_info.call_args_list, "run investigation_id is: pg_id"
```
Comment (dorschw) [inline demisto_sdk/tests/integration_tests/run_integration_test.py:178]:
```suggestion
        logger_debug.call_args_list, "running command in investigation_id=pg_id"
```

File: demisto_sdk/tests/integration_tests/run_integration_test.py [hunk 7]
```diff
@@ -141,3 +141,41 @@ def test_json_to_outputs_flag_fail_no_prefix(
...
+    Given
+    - kl-get-components command and --incident-id argument.
+
+    When
+    - Running `run` command on it.
+
+    Then
+    - Ensure the investigation-id is set from the incident-id.
+    """
+    logger_info = mocker.patch.object(logging.getLogger("demisto-sdk"), "info")
```
Comment (dorschw) [inline demisto_sdk/tests/integration_tests/run_integration_test.py:157]:
```suggestion
    logger_debug = mocker.patch.object(logging.getLogger("demisto-sdk"), "debug")
```

**Review by thefrieddan1 — APPROVED** [review]

**Pull request merged**